File: 2011-final-value-of-variable-after-performing-operations.cpp
/*
2011. Final Value of Variable After Peforming Operations

Submitted: November 21, 2024

Runtime: 0 ms (beats 100.00%)
Memory: 18.14 MB (beats 12.42%)
*/

class Solution {
public:
    int finalValueAfterOperations(vector<string>& operations) {
        signed char result = 0; // constraints limit length of operations to 100
        for (const string& op : operations) {
            if (op == "--X" || op == "X--") --result;
            else if (op == "++X" || op == "X++") ++result;
        }
        return result;
    }
};

File: 2011-final-value-of-variable-after-performing-operations.py
"""
2011. Final Value of Variable After Performing Operations

Submitted: January 17, 2025

Runtime: 0 ms (beats 100.00%)
Memory: 17.88 MB (beats 21.09)
"""

class Solution:
    def finalValueAfterOperations(self, operations: List[str]) -> int:
        return sum((-1 if op[1] == '-' else 1) for op in operations)